api/util/apps.go

```diff
@@ -117,6 +117,22 @@ func (git GitAuth) Enabled() bool {
 	return false
 }
 
+func (git GitAuth) Valid() error {
+	if git.SSHPrivateKey != nil {
+		if *git.SSHPrivateKey == "" {
+			return fmt.Errorf("the SSH private key cannot be empty")
+		}
+	}
+
+	if git.Username != nil && git.Password != nil {
+		if *git.Username == "" && *git.Password == "" {
```

Comment:
I think a logical "or" might be better here as both, the username or the password, should not be empty?

Reply:
yup you are right, added it